File: duolingo_rum/Models/Word.cs
using System;
using System.Collections.Generic;

namespace duolingo_rum.Models;

public partial class Word
{
    public int Id { get; set; }

    public int LanguageId { get; set; }

    public int? TopicId { get; set; }

    public string Word1 { get; set; } = null!;

    public string Translation { get; set; } = null!;

    public string? Transcription { get; set; }

    public string? ExampleSentence { get; set; }

    public string? ExampleTranslation { get; set; }

    public int? FrequencyRank { get; set; }

    public DateTime? CreatedAt { get; set; }

    public virtual ICollection<ExerciseResult> ExerciseResults { get; set; } = new List<ExerciseResult>();

    public virtual Language Language { get; set; } = null!;

    public virtual Topic? Topic { get; set; }

    public virtual ICollection<WordProgress> WordProgresses { get; set; } = new List<WordProgress>();
}

File: android-app/app/src/main/java/com/linguaai/app/models/Word.kt
package com.linguaai.app.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Word(
    val id: Int = 0,
    @SerialName("language_id")
    val languageId: Int = 0,
    @SerialName("topic_id")
    val topicId: Int? = null,
    val word: String = "",
    val translation: String = "",
    val transcription: String? = null,
    @SerialName("example_sentence")
    val exampleSentence: String? = null,
    @SerialName("example_translation")
    val exampleTranslation: String? = null,
    @SerialName("frequency_rank")
    val frequencyRank: Int? = null,
    @SerialName("created_at")
    val createdAt: String? = null
)
